src/assets/Generator.Shared/MgmtPlaneOperationHelpers.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Concurrent;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal static class MgmtPlaneOperationHelpers
+    {
+        /// <summary>
+        /// Store the property values that are shared by current management plane RP.
+        /// </summary>
+        private static ConcurrentDictionary<string, object> _mgmtProperties = new ConcurrentDictionary<string, object>();
```

Comment:
Lets figure out where this caching belongs.  Need to sync with https://github.com/Azure/azure-sdk-for-net/pull/27079#discussion_r813451688

Reply:
Done